58 make a standard dashboard for mas results

EserSteffen opened the pull request.

SteffenEserAC committed:
first shot on mas dashboard

SteffenEserAC committed:
Merge branch '56-percentage-logging-in-environment-does-not-adjust-when-environment-offset-is-non-zero' into 58-make-a-standard-dashboard-for-mas_results

SteffenEserAC committed:
add changelog and make dashboard more compact

SteffenEserAC committed:
make dashboard run in separate process and start with communicator results and vis

SteffenEserAC committed:
make communicator results logging and add to dashboard

SteffenEserAC committed:
Make results logging common to all communicators and add subscription multiprocessing

SteffenEserAC committed:
update CHANGELOG.md

SteffenEserAC committed:
add incremental and live dashboard. Currently incremental load function for simulator seems broken

SteffenEserAC committed:
fixes in dashboard and improve code

SteffenEserAC committed:
revert room_mas example to normal

SteffenEserAC committed:
fixes in simulator.py

SteffenEserAC committed:
fix default filename for overwriting in agentlogger

SteffenEserAC committed:
fix default filename for overwriting in agentlogger

SteffenEserAC committed:
change port detection and remove buggy error handling from simulator viz. Improve CHANGELOG.md

SteffenEserAC committed:
fix some tests and adjust code for agent_logger.py file handling to be more in one place. ci examples are now executed with tempfile

SteffenEserAC committed:
support python 3.8 style type hints

SteffenEserAC committed:
support python 3.8 style type hints

SteffenEserAC committed:
Merge remote-tracking branch 'origin/58-make-a-standard-dashboard-for-mas_results' into 58-make-a-standard-dashboard-for-mas_results

# Conflicts:
#	agentlib/core/module.py

SteffenEserAC committed:
some adjustments to agentlogger results

SteffenEserAC committed:
fix live update should never be blocking

sarahleidolf commented:
Update readme for new installation requirements. Currently, the following error message is displayed:
`Failed to import launch_mas_dashboard. Ensure Dash and its dependencies are installed: pip install agentlib[interactive]`
Newest matplotlib version (3.10.5) is not compatible: Update requirements.txt

sarahleidolf commented:
Thanks for the very cool update. I think communications logging is a particularly good addition. One point: Is it possible to define the functions ```get_results_incremental(), load_from_file_incremental(), visualize_results()``` globally, instead of defining them individually for all agents? This results in a lot of visualization code within the communicator, etc.

SteffenEserAC committed:
multiple changes:
- deal with pydantic deprecation warning
- update README.md
- move some logging to debug level
- move communicator result handling to a separate file

SteffenEserAC committed:
restructure com logging to remove some if checks

SteffenEserAC committed:
update mqtt to new version

SteffenEserAC committed:
update mqtt to new version

SteffenEserAC committed:
update agentlogger filename behaviour. MAS does not provide filename anymore, instead it relies on agentloggers default. AgentLogger auto-increments existing file names, but only for default

Commit:
Merge remote-tracking branch 'origin/main' into 58-make-a-standard-dashboard-for-mas_results

# Conflicts:
#	CHANGELOG.md
#	agentlib/__init__.py
#	agentlib/modules/communicator/mqtt.py
#	agentlib/modules/utils/agent_logger.py
#	examples/multi-agent-systems/room_mas/room_mas.py

FWuellhorst reviewed (dismissed):
@EserSteffen looks really nice. The "live" option only sort of worked for me, but the example is also really fast.

Really cool feature, only minor comments.

SteffenEserAC committed:
delete some AI slop, put multiprocessing comm in same file

SteffenEserAC committed:
update multiprocessing to its in one file and uses a mixin to avoid config and code duplication

SteffenEserAC committed:
add tests for new comm logging

lwaer committed:
merge main into this branch and resolve conflicts

lwaer committed:
set with_dashboard = True in room_mas example unit test

lwaer commented on tests/test_examples.py:
```diff
@@ -116,21 +122,21 @@ def test_csv_data_source(self):
         )
         self._run_example_with_return(
             file="simulation//csv_data_source.py",
             func_name="run_example",
             with_plots=False,
             log_level=logging.FATAL,
-            extrapolation="repeat"
+            extrapolation="repeat",
         )
         self._run_example_with_return(
             file="simulation//csv_data_source.py",
             func_name="run_example",
             with_plots=False,
             log_level=logging.FATAL,
-            extrapolation="backwards"
+            extrapolation="backwards",
```
Done. CI tests successfully.

lwaer committed:
implemented tests for new comm functions
